ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessor.java

```diff
@@ -172,10 +173,30 @@ public BlockProcessingResult processBlock(
     LOG.trace("traceStartBlock for {}", blockHeader.getNumber());
     blockTracer.traceStartBlock(worldState, blockHeader, miningBeneficiary);
 
+    Optional<BlockAccessListBuilder> blockAccessListBuilder =
+        protocolSpec
+            .getBlockAccessListFactory()
+            .filter(
+                b ->
+                    protocolSpec
+                        .getEvm()
+                        .getEvmConfiguration()
+                        .worldUpdaterMode()
+                        .equals(WorldUpdaterMode.STACKED))
```

Comment:
we cannot move it to the protocol spec ? like that this check will be in one place

Reply:
Do you mean to add a getter `getWorldUpdaterMode` to `protocolSpec`?

Reply:
disable BAL in https://github.com/hyperledger/besu/blob/main/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolSpecBuilder.java#L363 when evmConfiguration is not working with StackedUpdater

Reply:
Done

Reply:
I think now we can remove all the is instance of StackedUpdater in the code

Reply:
No, we can't. It is used to NOT use transaction-level updater in journaled mode, regardless of whether BALs are active or not.